BookingsApi/BookingsApi/Controllers/WorkAllocationsController.cs

```diff
@@ -73,15 +73,39 @@ public async Task<IActionResult> AllocateHearingAutomatically(Guid hearingId)
         [ProducesResponseType((int)HttpStatusCode.NotFound)]
         public async Task<IActionResult> GetUnallocatedHearings()
         {
-            var query = new GetAllocationHearingsBySearchQuery(isUnallocated: true);
+            var today = DateTime.UtcNow; //provide a range (from today 1 year) for unallocated hearings rather than return all past and present.
+            var query = new GetAllocationHearingsBySearchQuery(isUnallocated: true, fromDate: today, toDate: today.AddYears(1));
             var results = await _queryHandler.Handle<GetAllocationHearingsBySearchQuery, List<VideoHearing>>(query);
 
             if (results.Count <= 0)
                 _logger.LogInformation("[GetUnallocatedHearings] Could not find any unallocated hearings");
             var response = results.Select(HearingToDetailsResponseMapper.Map).ToList();
             return Ok(response);
         }
- 
+         
+        /// <summary>
+        /// Get the allocated cso for the provided hearing Ids
+        /// </summary>
+        /// <param name="hearingIds">Hearing Reference ID array</param>
+        /// <returns>list of hearing Ids with the allocated cso</returns>
+        [HttpGet("allocation", Name = "GetAllocationsForHearings")]
+        [OpenApiOperation("GetAllocationsForHearings")]
+        [ProducesResponseType(typeof(IList<AllocatedCsoResponse>), (int)HttpStatusCode.OK)]
+        public async Task<IActionResult> GetAllocationsForHearings([FromQuery] Guid[] hearingIds)
+        {
+            var allocatedCsoResponse = new List<AllocatedCsoResponse>();
+            foreach (var hearingId in hearingIds)
+            {
+                var justiceUser = await _queryHandler
+                    .Handle<GetCsoAllocationByHearingIdQuery, JusticeUser>(new GetCsoAllocationByHearingIdQuery(hearingId));
```

Comment:
Could we tweak this query to take a list of hearing ids rather than a single one (`GetCsoAllocationByHearingIdsQuery`) and fetch the data in a single database hit?

Reply:
Hmm. Yeah that's fair. Was trying to keep the query simple, but probably makes sense to just have one call. I'll have to change the return type to a Hearing entity rather than JusticeUser and add some logic in the controller.

Reply:
one option is to change the handler to return a `Tuple<Guid, JusticeUser>`. That way we could return the hearing id alongside the justice user:


```
public async Task<Tuple<Guid, JusticeUser>> Handle(GetCsoAllocationByHearingIdQuery query)
        {
            var hearing = await _context.VideoHearings
                .Include(h => h.Allocations)
                    .ThenInclude(a => a.JusticeUser)
                    .ThenInclude(ju => ju.UserRole)
                .FirstOrDefaultAsync(x => x.Id == query.HearingId);

            return new Tuple<Guid, JusticeUser>(hearing.Id, hearing?.AllocatedTo);
        }
```

Reply:
it's a shame we can't use named arguments like `Task<(Guid hearingId, JusticeUser user)>`, I think you have to declare it as a concetete `Tuple<>` for it to work with generics

Reply:
Yeah, some good suggestions. I think query handlers should only really return database entities though. Too may types of dtos add too much abstraction to it I think. And it should be transformed in a service or controller with logic in a differnt model. I've made the query quite simple now anyway, so it just a GetAllocationHearings query now, which can probably be reused elsewhere if need be.